site/app/controllers/calendar/CalendarController.php

```diff
@@ -27,8 +32,179 @@ class CalendarController extends AbstractController {
      */
     public function viewCalendar(): WebResponse {
         $user = $this->core->getUser();
-        $gradeables_of_user = GradeableUtils::getAllGradeableListFromUserId($this->core, $user);
 
-        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables_of_user));
+        $calendar_messages = [];
+
+        $courses = $this->core->getQueries()->getCourseForUserId($user->getId());
+
+        $gradeables_of_user = GradeableUtils::getAllGradeableListFromUserId($this->core, $user, $courses, $calendar_messages);
+
+        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables_of_user, $courses, $calendar_messages));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/calendar")
+     */
+    public function viewCourseCalendar(): WebResponse {
+        $calendar_messages = [];
+        $user = $this->core->getUser();
+        $courses = $this->core->getQueries()->getCourseForUserId($user->getId());
+        $name = $this->core->getConfig()->getCourse();
+        // Initialize $oneCourse and get the specific course for the course calendar
+        $oneCourse = $courses[0];
```

Comment:
I thought about that but it isn't possible to view the course calendar page for a given course without being a part of any courses. At least, that's what I figured. I can of course set it to something else or check if the value exists.

Reply:
That seems reasonable.  I don't think there's a need to handle anything gracefully because we'd want an error to show up in the logs if someone was able to access it somehow.